repos/system_upgrade/el7toel8/actors/persistentnetnamesconfig/tests/test_persistentnetnamesconfig.py

```diff
@@ -0,0 +1,121 @@
+import os
+
+import pytest
+
+from leapp.libraries.actor import persistentnetnamesconfig
+from leapp.models import PersistentNetNamesFacts, PersistentNetNamesFactsInitramfs
+from leapp.models import RenamedInterface, RenamedInterfaces, InitrdIncludes
+from leapp.models import Interface, PCIAddress
+from leapp.snactor.fixture import current_actor_context
+
+
+def generate_link_file_mocked(interface):
+    return '/etc/systemd/network/10-leapp-{}.link'.format(interface.name)
+
+
+def interface_mocked(i=0):
+    return Interface(
+        name='n{}'.format(i),
+        devpath='dp{}'.format(i),
+        driver='d{}'.format(i),
+        vendor='v{}'.format(i),
+        pci_info=PCIAddress(
+            domain='pd{}'.format(i),
+            bus='pb{}'.format(i),
+            function='pf{}'.format(i),
+            device='pd{}'.format(i)
+        ),
+        mac='m{}'.format(i)
+    )
+
+
+def generate_interfaces(count):
+    return [interface_mocked(i) for i in range(count)]
```

Comment:
just a note, it is usually might be a good idea to use (i for i in itereable) rather then list comprehension. In a first version it returns a generator

Reply:
Ah, finally understood. I thought you meant [this] returns a generator, rather than (this). Thanks!

Reply:
Sadly the generator doesn't work here, as I'm overwriting the list items later on (line 64, for example). Moreover, I'm using the list twice so the generator would generate the list twice, if I get it right.
Still, it's nice to know this technique.